tidy3d/components/lumped_element.py

```diff
@@ -107,24 +140,148 @@ def _voltage_axis_in_plane(cls, val, values):
+        """Converts the :class:`LumpedResistor` object to a :class:`Structure`
+        ready to be added to the :class:`Simulation`"""
+        conductivity = self._sheet_conductance
+        medium_dict = {
+            "tt": Medium(conductivity=conductivity),
+            "ss": Medium(conductivity=conductivity),
+        }
+        return Structure(
+            geometry=Box(size=self.size, center=self.center),
+            medium=Medium2D(**medium_dict),
+        )
 
-    resistance: pydantic.PositiveFloat = pydantic.Field(
+
+class CoaxialLumpedResistor(AbstractLumpedResistor):
+    """Class representing a coaxial lumped resistor. Lumped resistors are appended to the list of
+    structures in the simulation as :class:`Medium2D` with the appropriate conductivity given their
+    size and geometry."""
+
+    center: Coordinate = pydantic.Field(
+        (0.0, 0.0, 0.0),
+        title="Center",
+        description="Center of object in x, y, and z.",
+        units=MICROMETER,
+    )
+
+    outer_diameter: pydantic.PositiveFloat = pydantic.Field(
         ...,
-        title="Resistance",
-        description="Resistance value in ohms.",
-        unit=OHM,
+        title="Outer Diameter",
+        description="Diameter of the outer concentric circle.",
+        units=MICROMETER,
+    )
+
+    inner_diameter: pydantic.PositiveFloat = pydantic.Field(
+        ...,
+        title="Inner Diameter",
+        description="Diameter of the inner concentric circle.",
+        units=MICROMETER,
+    )
+
+    normal_axis: Axis = pydantic.Field(
+        ...,
+        title="Normal Axis",
+        description="Specifies the normal axis, which defines"
+        "the orientation of the circles making up the coaxial lumped element.",
     )
 
+    def to_mesh_overrides(self) -> List[MeshOverrideStructure]:
+        """Creates a suggested :class:`MeshOverrideStructure` list that could be added to the
+        :class:`Simulation`"""
+        # Create a mesh override when refinement is needed.
+        # An important use case is when a LumpedResistor is used with a LumpedPort
+        # The port is a flat surface, but when computing the port current,
+        # we'll eventually integrate the magnetic field just above and below
+        # this surface, so the mesh override needs to ensure that the mesh
+        # is fine enough not only in plane, but also in the normal direction.
+        # So in the normal direction, we'll make sure there are at least
+        # 2 cell layers above and below whose size is the same as the in-plane
+        # cell size in the override region. Also, to ensure that the port itself
+        # is aligned with a grid boundary in the normal direction, two separate
+        # override regions are defined, one above and one below the analytical
+        # port region.
+        mesh_overrides = []
+        size = 3 * [self.outer_diameter]
+        size[self.normal_axis] = 0
+        if self.num_grid_cells:
```

Comment:
It seems like a lot of this logic could be simplified with the `pop_axis()` method, or at least assigning some variables up front. For example `override_dl[self.normal_axis]` is referenced a lot. 

```py
size = unpop_axis(0, (self.outer_diameter, self.outer_diameter), axis=self.normal_axis)
```

```py
dl = self.outer_diameter / self.num_grid_cells
override_dl = pop_axis(dl/2, (dl, dl), axis=self.normal_axis)
```
etc.

Up to you whether you think this makes things more complex, but in my read, it might be more declarative and easier to read.